### DISCUSSION TITLE
Leave Room button does not work when embedded on other websites HC

### DISCUSSION THREAD

**Discussion by normanwankowski** [mozilla/hubs#5220] [Hubs Cloud Questions]
Hi I have an issue with Hubs Cloud. I embedded a room on my website. When I hit the LEAVE ROOM button I cannot leave the room, nothing is happening. Dunno why. The button script, refresh content is not working for embeds on external sites I think. How to fix?
Expected behavior: when clicking on LEAVE ROOM the embedded room should refresh itself to the initial state. Instead the button is reloading the homepage of the original domain.
Can someone please embed a room on a website and test it. It is a huge problem for me. Because my clients can't embed the Hubs Clouds rooms.

<img width="861" alt="Bildschirmfoto 2022-03-07 um 14 06 22" src="https://user-images.githubusercontent.com/92306028/157040085-482a1678-c8d2-47b3-ab81-1d388019774c.png">

**Comment by pattersonbl2** [comment]
Do you have any browser console logs when the button is pressed?   I was unable to reproduce this issue.

**Reply by normanwankowski** [reply]
@Pattersonbl2 Hi Brandon, thanks for looking into that. The site is online you can test it yourself with this link:  [](

https://www.kawaii.com/metaverse

I checked the console log but I couldn't really find something. When I hit the Leave Room button simply nothing happens.
![Bildschirmfoto 2022-03-09 um 12 52 15](https://user-images.githubusercontent.com/92306028/157436957-9fdd7f71-03ee-4264-9b59-6aeb186bfcca.png)

**Reply by pattersonbl2** [reply]
After doing a little research, I saw the issue is related to the X-Frame option. This is set as same orgian, which is disallowed.  More info can be found [here](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Frame-Options).  I am not sure how you would fix this unless you take away the iframe the website is wrapped in.

**Reply by normanwankowski** [reply]
@Pattersonbl2 Oh I see. Thanks for the research. But this must be related to my BC settings on AWS right? Because I checked the iframe code als on other sites (webflow, html online editor, etc.) and on all these sites the Leave Room button doesn't work.

I used this embed code:

`<iframe src="https://tmvr.io/hzodb6E?embed_token=f9a9930d5c465b8603535b53321b022c" style="width: 1024px; height: 768px;" allow="microphone; camera; vr; speaker;"></iframe>`


Do you have an idea where I can change the X-Frame setting in my Cloud Formation template for HC?
